onadata/libs/tests/utils/test_export_tools.py

```diff
@@ -261,6 +262,44 @@ def test_get_value_or_attachment_uri(self):
         self.assertTrue(val_or_url)
         self.assertEqual(value, val_or_url)
 
+    def test_get_attachment_uri_for_filename_with_space(self):
+        path = os.path.join(
+            os.path.dirname(__file__), 'fixtures',
+            'photo_type_in_repeat_group.xlsx')
+        self._publish_xls_file_and_set_xform(path)
+
+        filename = u'bob/attachments/1 2 3.jpg'
```

Comment:
`filename` should actually be `bob/attachments/1_2_3.jpg` to properly address the issue.